spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -122,7 +126,48 @@ def refresh_from_pdb(self, pdb_state):
         if 'var_properties' in pdb_state:
             self.set_var_properties(pdb_state['var_properties'])
 
+    def set_pdb_state(self, pdb_state):
+        """Set current pdb state."""
+        if pdb_state is not None and isinstance(pdb_state, dict):
+            self.refresh_from_pdb(pdb_state)
+
+    def pdb_continue(self):
+        """Continue debugging."""
+        # Run Pdb continue to get to the first breakpoint
+        # Fixes 2034
+        self.pdb_execute('continue')
+
     # ---- Private API (overrode by us) ----------------------------
+    def _readline_callback(self, line):
+        """Callback used when the user inputs text in stdin."""
+        if not self.is_waiting_pdb_input():
+            if self._input_ready:
+                # This is a regular input call
+                self._input_ready = False
+                self._reading = False
+                self._append_before_prompt_cursor.setPosition(
+                    self._get_end_cursor().position())
+                return self.kernel_client.input(line)
+            else:
+                # This is an error. Raise?
+                return
+        line = line.strip()
+
+        # Save history to browse it later
+        self._pdb_line_num += 1
+        self.add_to_pdb_history(self._pdb_line_num, line)
+
+        if line:
+            self._last_pdb_cmd = line
+
+        # This is the Spyder addition: add a %plot magic to display
+        # plots while debugging
+        if line.startswith('%plot '):
+            line = line.split()[-1]
+            line = "__spy_code__ = get_ipython().run_cell('%s')" % line
```

Comment:
I am not sure what this code does. Should we keep it? What is it supposed to do?

Reply:
It's used to generate inline plots while debugging by calling this `%plot` pseudo-magic I added to our debugger some time ago. Without this trick, plots can't be seen while debugging.

If you think this can implemented better, please leave it for a another PR.